src/Orleans/Serialization/TypeUtilities.cs

```diff
@@ -80,6 +80,11 @@ internal static bool IsOrleansShallowCopyable(this Type t)
             return false;
         }
 
+        private static bool IsTypeFieldsShallowCopyable(TypeInfo typeInfo)
+        {
+            return typeInfo.GetFields().All(f => !(f.FieldType.IsEquivalentTo(typeInfo)) && IsOrleansShallowCopyable(f.FieldType));
```

Comment:
Sorry, to be completely accurate, it must be changed to:
```
typeInfo.GetFields().All(f => !f.FieldType.GetTypeInfo().IsEquivalentTo(typeInfo.AsType()) && IsOrleansShallowCopyable(f.FieldType))
```
which might default the purpose of this PR. The point is that it's currently not portable and we'll have to change it